### PR TITLE
Convert TOC to YAML

### LINKED ISSUE #9055: Convert ToCs to YAML format
Migrate the ASP.NET 4.x and ASP.NET Core ToCs to YAML format. Development has stopped on the Markdown format we use.

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #9055

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@tdykstra 	TOC doesn't show up in the [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/?branch=pr-en-us-10498&view=aspnetcore-2.2)

File: aspnetcore/toc.yml [hunk 1]
```diff
@@ -0,0 +1,807 @@
...
+              href: uid:tutorials/razor-pages/da1
+            - name: Add search
+              href: uid:tutorials/razor-pages/search
+            - name: Add a new field
+              href: uid:tutorials/razor-pages/new-field
+            - name: Add validation
+              href: uid:tutorials/razor-pages/validation
+    - name: MVC
+      items: 
+        - name: MVC Overview
```
Comment (scottaddie) [inline aspnetcore/toc.yml]:
```suggestion
        - name: Overview
```

File: aspnetcore/toc.yml [hunk 2]
```diff
@@ -0,0 +1,807 @@
...
+          href: uid:tutorials/signalr
+        - name: SignalR with TypeScript
+          href: uid:tutorials/signalr-typescript-webpack
+    - name: Samples
+      href: https://github.com/aspnet/SignalR-samples
+    - name: Server concepts
+      items: 
+        - name: Hubs
+          href: uid:signalr/hubs
+        - name: HubContext
```
Comment (scottaddie) [inline aspnetcore/toc.yml]: I suggest using a different name. The name `HubContext` is meaningless to those who've never used the feature.

File: aspnetcore/toc.yml [hunk 3]
```diff
@@ -0,0 +1,807 @@
...
+          href: uid:signalr/api-design
+    - name: Clients
+      items: 
+        - name: .NET client
+          href: uid:signalr/dotnet-client
+        - name: .NET API reference
+          href: /dotnet/api/microsoft.aspnetcore.signalr.client
+        - name: Java client
+          href: uid:signalr/java-client
+        - name: Java API reference
```
Comment (scottaddie) [inline aspnetcore/toc.yml]: The Java API is available only in ASP.NET Core 2.2 and above. Is there a way to toggle visibility accordingly in this YAML-based ToC?

File: aspnetcore/toc.yml [hunk 4]
```diff
@@ -0,0 +1,807 @@
...
+          href: uid:security/authorization/resourcebased
+        - name: View-based authorization
+          href: uid:security/authorization/views
+        - name: Limit identity by scheme
+          href: uid:security/authorization/limitingidentitybyscheme
+    - name: Data protection
+      items: 
+        - name: Overview
+          href: uid:security/data-protection/introduction
+        - name: Data Protection APIs
```
Comment (scottaddie) [inline aspnetcore/toc.yml]:
```suggestion
        - name: Data protection APIs
```

File: aspnetcore/toc.yml [hunk 5]
```diff
@@ -0,0 +1,807 @@
...
+          href: uid:performance/caching/response
+        - name: In-memory cache
+          href: uid:performance/caching/memory
+        - name: Distributed caching
+          href: uid:performance/caching/distributed
+        - name: Response caching middleware
+          href: uid:performance/caching/middleware
+    - name: Response compression
+      href: uid:performance/response-compression
+    - name: Diagnostic Tools
```
Comment (scottaddie) [inline aspnetcore/toc.yml]:
```suggestion
    - name: Diagnostic tools
```

File: aspnetcore/toc.yml [hunk 6]
```diff
@@ -0,0 +1,807 @@
...
+  items: 
+    - name: Overview
+      href: uid:fundamentals/localization
+    - name: Portable Object localization
+      href: uid:fundamentals/portable-object-localization
+- name: Advanced
+  items: 
+    - name: URL rewriting
+      href: uid:fundamentals/url-rewriting
+    - name: File Providers
```
Comment (scottaddie) [inline aspnetcore/toc.yml]:
```suggestion
    - name: File providers
```

File: aspnetcore/toc.yml (hunk 3 above)
Comment (tdykstra) [reply aspnetcore/toc.yml]: So far as I know that only works by setting moniker range in article metadata, i.e., it's not a Markdown vs. YAML difference.

**Comment by tdykstra** [comment]
Replaced by #10603

**Pull request closed without merging**